jira/client.py

```diff
@@ -720,7 +723,7 @@ def _fetch_pages(
             page_params["startAt"] = startAt
         if maxResults:
             page_params["maxResults"] = maxResults
-        elif batch_size is not None:
+        elif batch_size := self._get_batch_size(item_type):
             page_params["maxResults"] = batch_size
```

Comment:
```suggestion
        else:
            page_params["maxResults"] = self._get_batch_size(item_type)
```

otherwise won't correctly apply when the batch size is configured to None

Probably a need a test for that

Reply:
@adehad, I tried before and apparently the API wont accept "maxResults=null". That's why I specifically only set it when its truthy, so we ignore `None`. Am I mistaken or missing something; for the old maxResults=False we did not set `maxResults` either?


---
<sub>Jannik Meinecke (<jannik.meinecke@mercedes-benz.com>) on behalf of MBition GmbH.
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)</sub>